airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -331,32 +331,41 @@ def read_records(
-                if properties_list:
-                    stream_records, response = self._read_stream_records(
-                        properties_list=properties_list,
-                        stream_slice=stream_slice,
-                        stream_state=stream_state,
-                        next_page_token=next_page_token,
-                    )
-                    records = [value for key, value in stream_records.items()]
-                else:
-                    response = self.handle_request(stream_slice=stream_slice, stream_state=stream_state, next_page_token=next_page_token)
-                    records = self._transform(self.parse_response(response, stream_state=stream_state, stream_slice=stream_slice))
-
-                if self.filter_old_records:
-                    records = self._filter_old_records(records)
-                yield from records
-
-                next_page_token = self.next_page_token(response)
-                if not next_page_token:
-                    pagination_complete = True
-
-            # Always return an empty generator just in case no records were ever yielded
-            yield from []
+        try:
+            with AirbyteSentry.start_transaction("read_records", self.name), AirbyteSentry.start_transaction_span("read_records"):
+                while not pagination_complete:
+
+                    properties_list = list(self.properties.keys())
+                    if properties_list:
+                        stream_records, response = self._read_stream_records(
+                            properties_list=properties_list,
+                            stream_slice=stream_slice,
+                            stream_state=stream_state,
+                            next_page_token=next_page_token,
+                        )
+                        records = [value for key, value in stream_records.items()]
+                    else:
+                        response = self.handle_request(
+                            stream_slice=stream_slice, stream_state=stream_state, next_page_token=next_page_token
+                        )
+                        records = self._transform(self.parse_response(response, stream_state=stream_state, stream_slice=stream_slice))
+
+                    if self.filter_old_records:
+                        records = self._filter_old_records(records)
+                    yield from records
+
+                    next_page_token = self.next_page_token(response)
+                    if not next_page_token:
+                        pagination_complete = True
+
+                # Always return an empty generator just in case no records were ever yielded
+                yield from []
+        except requests.exceptions.HTTPError as e:
+            status_code = e.response.status_code
+            if status_code == 403:
+                raise RuntimeError("Invalid permissions. Please ensure the all scopes are authorized for.")
```

Comment:
we should say the stream name that broke here just so the user has actionable feedback